projects/batfish/src/main/java/org/batfish/representation/juniper/BgpGroup.java

```diff
@@ -72,6 +79,15 @@ public final void cascadeInheritance() {
       if (_advertisePeerAs == null) {
         _advertisePeerAs = _parent._advertisePeerAs;
       }
+      if (_authenticationAlgorithm == null) {
+        _authenticationAlgorithm = _parent._authenticationAlgorithm;
+      }
+      if (_authenticationKey == null) {
+        _authenticationKey = _parent._authenticationKey;
+      }
+      if (_authenticationKeyChainName == null) {
+        _authenticationKeyChainName = _parent._authenticationKeyChainName;
+      }
```

Comment:
One thing I am not sure: what if a BgpGroup specified authentication keychain and algorithm,(by authentication-key-chain and authentication-algorithm commands), and its parent BgpGroup has authentication-key specified, should the child get the `authentication key` from its parent?